lib/attachments/gridstore-storage.js

```diff
@@ -137,6 +139,25 @@ class GridstoreStorage {
                             // might be already finished if retrying after delay
                             return;
                         }
+                        if (args.length > 2) {
+                            const calculatedFileContentHash = args[2];
+
+                            if (calculatedFileContentHash) {
+                                this.gridfs.collection(this.bucketName + '.files').findOneAndUpdate(
+                                    {
+                                        _id: hash
+                                    },
+                                    {
+                                        $set: {
+                                            'metadata.fileContentHash': calculatedFileContentHash
+                                        }
+                                    },
+                                    {
+                                        returnDocument: 'after'
+                                    }
+                                );
+                            }
+                        }
                         callback(...args);
```

Comment:
There's a potential race condition if we want to use the sha256 hash immediately after the attachment has been uploaded - there's a chance that the `findOneAndUpdate` call is not finished yet, and thus, the value is not available.